feat(gcp): add compute_instance_group_manager

kevinkengne opened the pull request:
**Objective:**

Add support for `compute_instance_group_manager` and `compute_region_instance_group_manager`.

**Status:**

- [x] Added to registry.go
- [x] Added resource file
- [x] Added integration tests
- [x] Added unit tests if applicable - Not applicable

**Issues:**

Currently as suggested by @tim775, only one apllication `version` is supported per instance group. Support for multiple version per instance group could be added in a separate PR.

**Useful links:**

Pricing: 

https://cloud.google.com/compute/vm-instance-pricing
https://cloud.google.com/compute/disks-image-pricing#disk
https://cloud.google.com/compute/gpus-pricing

Terraform: 

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance_group_manager
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_region_instance_group_manager
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance_template

tim775 commented on internal/providers/terraform/google/compute_instance_group_manager.go:
```diff
@@ -0,0 +1,116 @@
+
+func GetComputeInstanceGroupManagerRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:                "google_compute_instance_group_manager",
+		RFunc:               NewComputeInstanceGroupManager,
+		ReferenceAttributes: []string{"version.0.instance_template"},
```
```suggestion
		Name:                "google_compute_instance_group_manager",
		RFunc:               NewComputeInstanceGroupManager,
		Notes:               []string{"Multiple versions are not supported."},
		ReferenceAttributes: []string{"version.0.instance_template"},
```

tim775 commented on internal/providers/terraform/google/compute_region_instance_group_manager.go:
```diff
@@ -0,0 +1,38 @@
+
+func GetComputeRegionInstanceGroupManagerRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:                "google_compute_region_instance_group_manager",
+		RFunc:               NewComputeRegionInstanceGroupManager,
+		ReferenceAttributes: []string{"version.0.instance_template"},
```
```suggestion
		Name:                "google_compute_region_instance_group_manager",
		RFunc:               NewComputeRegionInstanceGroupManager,
		Notes:               []string{"Multiple versions are not supported."},
		ReferenceAttributes: []string{"version.0.instance_template"},
```

tim775 reviewed (changes requested):
This is great Kevin, thanks!  Looks like it turned into a bit of a beast.  I added a couple quick suggestions to note that we don't support multiple versions.

kevinkengne commented:
> This is great Kevin, thanks! Looks like it turned into a bit of a beast. I added a couple quick suggestions to note that we don't support multiple versions.

Thanks Tim! I've added the changes
